core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -275,7 +294,23 @@ public HttpFeed() {
     protected HttpFeed(Builder builder) {
         setConfig(ONLY_IF_SERVICE_UP, builder.onlyIfServiceUp);
         Map<String,String> baseHeaders = ImmutableMap.copyOf(checkNotNull(builder.headers, "headers"));
-        
+
+        HttpExecutor httpExecutor;
+        if (builder.httpExecutor != null) {
+            httpExecutor = builder.httpExecutor;
+        } else {
+            HttpExecutorFactory httpExecutorFactory = null;
+            Collection<? extends Location> locations = Locations.getLocationsCheckingAncestors(builder.entity.getLocations(), builder.entity);
+            Maybe<Location> location =  Machines.findUniqueElement(locations, Location.class);
```

Comment:
Better get the `SshMachineLocation` here (there might or might not be another location) and get the config from it. Inheritance should take care of getting the config from the right location.